support-frontend/app/controllers/DigitalSubscriptionFormController.scala

```diff
@@ -0,0 +1,122 @@
+package controllers
+
+import actions.CustomActionBuilders
+import admin.settings.{AllSettings, AllSettingsProvider, SettingsSurrogateKeySyntax}
+import assets.{AssetsResolver, RefPath, StyleContent}
+import cats.implicits._
+import com.gu.identity.model.{User => IdUser}
+import com.gu.monitoring.SafeLogger
+import com.gu.monitoring.SafeLogger._
+import com.gu.support.catalog.DigitalPack
+import com.gu.support.config.{PayPalConfigProvider, StripeConfigProvider}
+import com.gu.support.pricing.PriceSummaryServiceProvider
+import com.gu.support.zuora.api.ReaderType.{Direct, Gift}
+import config.RecaptchaConfigProvider
+import controllers.UserDigitalSubscription.{redirectToExistingThankYouPage, userHasDigitalSubscription}
+import play.api.mvc._
+import play.twirl.api.Html
+import services._
+import views.EmptyDiv
+import views.html.helper.CSRF
+import views.html.subscriptionCheckout
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DigitalSubscriptionFormController(
+  priceSummaryServiceProvider: PriceSummaryServiceProvider,
+  val assets: AssetsResolver,
+  val actionRefiners: CustomActionBuilders,
+  identityService: IdentityService,
+  testUsers: TestUserService,
+  membersDataService: MembersDataService,
+  stripeConfigProvider: StripeConfigProvider,
+  payPalConfigProvider: PayPalConfigProvider,
+  components: ControllerComponents,
+  settingsProvider: AllSettingsProvider,
+  fontLoaderBundle: Either[RefPath, StyleContent],
+  recaptchaConfigProvider: RecaptchaConfigProvider
+)(implicit val ec: ExecutionContext) extends AbstractController(components) with SettingsSurrogateKeySyntax {
+
+  import actionRefiners._
+
+  implicit val a: AssetsResolver = assets
+
+  def displayForm(orderIsAGift: Boolean): Action[AnyContent] = {
+    implicit val settings: AllSettings = settingsProvider.getAllSettings()
+    if (settings.switches.enableDigitalSubGifting.isOn || !orderIsAGift) {
+      authenticatedAction(subscriptionsClientId).async { implicit request =>
+
+        identityService.getUser(request.user.minimalUser).fold(
+          error => {
+            SafeLogger.error(scrub"Failed to display digital subscriptions form for ${request.user.minimalUser.id} due to error from identityService: $error")
+            Future.successful(InternalServerError)
+          },
+          user =>
+            if (orderIsAGift)
+              Future.successful(Ok(digitalSubscriptionFormHtml(user, orderIsAGift = true)))
+            else for (alreadyADigitalSubscriber <- userHasDigitalSubscription(membersDataService, request.user)) yield
+              if (alreadyADigitalSubscriber)
+                redirectToExistingThankYouPage
+              else
+                Ok(digitalSubscriptionFormHtml(user, orderIsAGift = false))
+        ).flatten.map(_.withSettingsSurrogateKey)
```

Comment:
the only change I made here is to flip so we only check if someone is an existing subscriber when necessary.  Before, for gift, we used to check, then ignore the result.